internal/net/grpc/client.go

```diff
@@ -186,19 +191,22 @@ func (g *gRPCClient) StartConnectionMonitor(ctx context.Context) (<-chan error,
 									ech <- err
 								}
 							}
-							if err == nil {
+							if err == nil && addr != "" && p != nil {
```

Comment:
![badge](https://img.shields.io/badge/review-nits-green.svg?style=flat-square)
`addr != ""` is already checked at L182 (`len(addr) != 0`) and `addr` is not modified between L182 and L194.